tgui/packages/tgui/interfaces/Photocopier.tsx

```diff
@@ -0,0 +1,385 @@
+import { sortBy } from 'common/collections';
+import { useState } from 'react';
+
+import { BooleanLike } from '../../common/react';
+import { useBackend } from '../backend';
+import {
+  Button,
+  Input,
+  LabeledList,
+  ProgressBar,
+  Section,
+  Slider,
+  Stack,
+} from '../components';
+import { Window } from '../layouts';
+
+type Data = {
+  has_item: BooleanLike;
+  has_toner: BooleanLike;
+  can_AI_print: BooleanLike;
+  isAI: BooleanLike;
+  is_photo: BooleanLike;
+  categories: string[];
+  color_mode: string;
+  num_copies: number;
+  max_copies: number;
+  copies_left: number;
+  max_toner: number;
+  current_toner: number;
+  paper_count: number;
+  max_paper_count: number;
+  blanks: Blank[];
+};
+
+type Blank = {
+  name: string;
+  category: string;
+  code: string;
+};
+
+export const Photocopier = (props) => {
+  const [selectedBlank, setSelectedBlank] = useState('');
+  const [selectedCategory, setSelectedCategory] = useState('');
+
+  return (
+    <Window
+      title="Photocopier"
+      width={selectedCategory ? 550 : 225}
+      height={525}
+    >
+      <Window.Content>
+        <Stack fill>
+          <Stack.Item basis={selectedCategory ? '40%' : '100%'}>
+            <Stack fill vertical>
+              <Stack.Item>
+                <Status selectedBlank={selectedBlank} />
+              </Stack.Item>
+              <Stack.Item>
+                <Actions selectedBlank={selectedBlank} />
+              </Stack.Item>
+              <Stack.Item grow>
+                <Categories
+                  selectedCategory={selectedCategory}
+                  setSelectedCategory={setSelectedCategory}
+                />
+              </Stack.Item>
+            </Stack>
+          </Stack.Item>
+          {selectedCategory && (
+            <Stack.Item basis="60%">
+              <Stack fill vertical>
+                <Stack.Item grow>
+                  <Blanks
+                    selectedCategory={selectedCategory}
+                    selectedBlank={selectedBlank}
+                    setSelectedBlank={setSelectedBlank}
+                  />
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+          )}
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const Status = ({ selectedBlank }) => {
```

Comment:
prefer we don't get into this pattern bc not only is it `any` type but it's also assigned as mutable `let selectedBlank` when you destructure inline like this. `props: PropTypeName` pls

Reply:
sorry, I wasn't specific enough here. I mean inlining the destructure statement itself. this assigns it as `let` 
```tsx
const Categories = ({
  selectedCategory,
  setSelectedCategory,
}: {
  selectedCategory: string | null;
  setSelectedCategory: (category: string) => void;
}
```
would become
```tsx
type CategoryProps = {
  selectedCategory: string | null;
  setSelectedCategory: (category: string) => void;
}

const Categories = (props: CategoryProps) => {
  const { selectedCategory, setSelectedCategory } = props;
```

Reply:
Oh.. well, thanks, i'll do it tomorrow <3
Just looked at other UI's, and it was like i did